s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/WMSOverlayRepository.java

```diff
@@ -10,11 +11,42 @@
 
 @Transactional(readOnly = true)
 public interface WMSOverlayRepository extends CrudRepository<WMSOverlay, Long> {
-    <T> Optional<T> findByOwnerTypeAndId(OverlayOwner owner, Long id, Class<T> projection);
-    <T> Optional<T> findByOwnerTypeAndIdAndInstitutionId(OverlayOwner owner, Long id, Long institutionId, Class<T> projection);
-    <T> Optional<T> findByOwnerTypeAndIdAndAppUserId(OverlayOwner owner, Long id, Long appUserId, Class<T> projection);
+    <T> Optional<T> findByIdAndAppUserIdAndOwnerType(
+            Long id,
+            Long appUserId,
+            OverlayOwner ownerType,
+            Class<T> projection
+    );
+    <T> Optional<T> findByIdAndOwnerTypeAndInstitutionId(
+            Long id,
+            OverlayOwner ownerType,
+            Long institutionId,
+            Class<T> projection
+    );
+    void deleteByIdAndOwnerType(Long id, OverlayOwner owner);
 
-    <T> List<T> findAllByOwnerType(OverlayOwner owner, Class<T> projection);
-    <T> List<T> findAllByOwnerTypeAndInstitutionId(OverlayOwner owner, Long institutionId, Class<T> projection);
-    <T> List<T> findAllByOwnerTypeAndAppUserId(OverlayOwner owner, Long appUserId, Class<T> projection);
+    List<WMSOverlay> findAllByOwnerType(OverlayOwner ownerType);
+
+    @Query(value = "SELECT wms_overlay.*\n" +
```

Comment:
@Query("SELECT w " +
            "FROM WMSOverlay w " +
            "LEFT JOIN FETCH w.institution i " +
            "LEFT JOIN FETCH i.membersRoles r " +
            "LEFT JOIN FETCH r.user u " +
            "WHERE w.ownerType = :ownerType AND u.id = :appUserId AND r.role= :role")
    List<WMSOverlay> findAllInstitutional(Long appUserId, AppRole role, OverlayOwner ownerType);

    @Query("SELECT w " +
            "FROM WMSOverlay w " +
            "WHERE w.appUser.id = :appUserId AND w.ownerType = :ownerType")
    List<WMSOverlay> findAllPersonal(Long appUserId, OverlayOwner ownerType);